lib/spack/spack/cmd/edit.py

```diff
@@ -121,22 +118,42 @@ def locate_file(name: str, path: str) -> str:
     return files[0]
 
 
+def repo_from_args(args):
```

Comment:
I think putting this in a function is not much of an improvement, cause now it has a return type of `Optional[Union[spack.repo.Repo, spack.repo.RepoPath]]`, and in the single place where it's used, it makes use of properties that exist only in `spack.repo.Repo`, and not `spack.repo.RepoPath`.

If you add type hints, you basically have to repeat all the branches in the call site.